### ISSUE TITLE
check_subrc: False positive when after else/endif

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#1723]
/zfoobar.prog.abap[4, 3]: Check sy-subrc(check_subrc)

```abap
REPORT zfoobar.
DATA ls_t100 TYPE t100.
IF 1 = 1.
  UPDATE t100 FROM @ls_t100. "<<< check_subrc error
ELSE.
  WRITE: 'BAD'.
ENDIF.
IF sy-subrc = 0.  "<<< this is the next statement after UPDATE
  WRITE: 'OK'.
ENDIF.
```

**Comment by larshp** [comment]
I'm not sure about this specific example, there is one UPDATE, so the subrc check should be moved inside the IF

However, I think there are good examples for this scenario/bug

**Comment by mbtools** [comment]
I reduced it to a minimum. The real case is actually quite common, I think:

```abap
SELECT...
IF sy-subrc = 0.
  UPDATE ...
ELSE.
  INSERT ...
ENDIF.
IF sy-subrc = 0.
  MESSAGE ...
ENDIF.
```

**Comment by larshp** [comment]
yeap 👍

**Comment by mbtools** [comment]
also with case:

```abap
REPORT zfoobar.
CASE sy-index.
  WHEN 1.
    FIND 'TEST1' IN TABLE lt_code RESULTS lt_results. "<<< check_subrc error
  WHEN 2.
    FIND 'TEST2' IN TABLE lt_code RESULTS lt_results. "<<< check_subrc error
ENDCASE.
IF sy-subrc = 0. "<<<  next statement
  WRITE 'OK'.
ENDIF.
```
